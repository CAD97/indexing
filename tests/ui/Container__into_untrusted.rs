use windex::scope_val;

fn main() {
    let v = vec![0];
    scope_val(v, |mut v| {
        let _ix = v.vet(0).unwrap();
        let r = v.as_ref_mut().into_untrusted();
        r.clear();
        // ix is now invalid logically but not statically
    })
}

// Remember to update the actual doc when this test changes!
