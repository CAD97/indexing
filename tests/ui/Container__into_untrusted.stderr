error[E0597]: `v` does not live long enough
  --> $DIR/Container__into_untrusted.rs:7:17
   |
5  |     scope_val(v, |mut v| {
   |                   ----- has type `windex::container::Container<'1, std::vec::Vec<i32>>`
6  |         let _ix = v.vet(0).unwrap();
7  |         let r = v.as_ref_mut().into_untrusted();
   |                 ^-------------
   |                 |
   |                 borrowed value does not live long enough
   |                 argument requires that `v` is borrowed for `'1`
...
10 |     })
   |     - `v` dropped here while still borrowed

For more information about this error, try `rustc --explain E0597`.
